src/bip32/derivation_path.rs

```diff
@@ -0,0 +1,121 @@
+use std::str::FromStr;
+
+use super::child_number::{ChildNumber, ChildNumberError};
+
+const PATH_PREFIX: &str = "m/";
+pub const MAX_DEPTH: usize = 5;
+
+#[derive(Clone, Copy, Debug)]
+pub enum Error {
+    MaxDepthExceeded,
+    WrongPathPrefix,
+    InvalidPathIndex(ChildNumberError),
+    CannotParseindex
+}
+
+pub struct DerivationPath {
+    pub path: Vec<ChildNumber>
```

Comment:
It would be good if we explicitly set the depth of the derivation path at compile time so that everywhere we know the depth at compile time.

Something like that.  (we also need to implements other methods like push, len and so on).

```
pub struct KeyPathVec<T, const MAX: usize> {
    inner: Vec<T>,
}

impl<T, const MAX: usize> KeyPathVec<T, MAX> {
    pub fn new() -> Self {
        Self { inner: Vec::new() }
    }
}
```